### PR TITLE
Frontend sign up form

### LINKED ISSUE #23: [Frontend] Sign Up Form Flow
# Description

We are gonna continue extending the user sign-up/log-in flow. When the user clicks "Sign up here" in the log-in page, it should bring them to the sign-up form that's found in "onboarding" under "READY FOR DEV" on the Figma. We are only focusing on the frontend for this case, as the backend is being done on another ticket. If all the fields are filled out, then send the user to the "Successful" page.

Screen Shot 2024-10-31 at 1 36 39 PM

# Testing
Not too much to test, but make sure all the individual parts of the form are working as intended, which could get confusing just because the form is so large and there's so much information the user needs to fill in. Edge cases could be highlighting required fields with a red border if they aren't filled in when the user tries to submit.

### PR DESCRIPTION
**Pull request by wkim10**
# Description

Created sign up form

## Issues

Resolve #23 #38 

## Screenshots

Screenshot 2024-12-11 at 4 10 59 AM

## Test

Try creating multiple users, omit certain fields + mess up certain fields to test edge cases

## Possible Downsides

Might need to remove the "volunteerType" field from the schema later

## Additional Documentations

None :)

### CODE REVIEW

File: src/components/SignUp.tsx [hunk 1]
```diff
@@ -0,0 +1,534 @@
...
+      address,
+      isOverAge14,
+      isFirstTime,
+      hasDriverLicense,
+      speakSpanish,
+      why,
+      comments,
+    };
+
+    console.log(formData);
```
Comment (johnny-t06) [inline src/components/SignUp.tsx]: nit: remove console logs

**Review by johnny-t06 — APPROVED** [review]
LGTM, great job guys looks amazing! Tested and works well
